src/tests/multihost/alltests/test_hosts.py

```diff
@@ -237,3 +237,38 @@ def test_009_ipnetwork_iphost(self, multihost):
             status = 'FAIL'
         client.sssd_conf('domain/example1', domain_params, action='delete')
         assert status == 'PASS'
+
+    @pytest.mark.tier1_3
+    def test_more_than_one_cn(self, multihost):
+        """
+        :title: hosts: 'getent hosts' not return hosts
+         if they have more than one CN in LDAP
+        :bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=2056035
+        :id: f2bacc70-ccff-11ec-9b25-845cf3eff344
+        :customerscenario: true
+        :steps:
+          1. Create hostname having multiple 'cn'
+          2. Entry having multiple 'cn' handled correctly by SSSD
+        :expectedresults:
+          1. Should succeed
+          2. Should succeed
+        """
+        ldap_uri = 'ldap://%s' % (multihost.master[0].sys_hostname)
+        ds_rootdn = 'cn=Directory Manager'
+        ds_rootpw = 'Secret123'
+        ldap_inst = LdapOperations(ldap_uri, ds_rootdn, ds_rootpw)
+        user_info = {'cn': 'node4'.encode('utf-8'),
+                     'cn': 'node4.example.test'.encode('utf-8'),
```

Comment:
cn is not getting overwrite, but changes done


dn: cn=node4+ipHostNumber=192.168.1.4,ou=People,dc=example,dc=test
cn: node4.example.test
cn: node4
objectClass: top
objectClass: ipHost
objectClass: device
ipHostNumber: 192.168.1.4

Reply:
Hi,

it is working because 389ds ensures that the `cn` from the `dn` is present in the object as well. Effectively `user_info` will only contain `cn: node4.example.test`.

```
$ python
Python 3.10.5 (main, Jun  9 2022, 00:00:00) [GCC 11.3.1 20220421 (Red Hat 11.3.1-2)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> user_info = {'cn': 'node4'.encode('utf-8'),
... 'cn': 'node4.example.test'.encode('utf-8'),
... 'ipHostNumber': '192.168.1.4'.encode('utf-8')}
>>> print (user_info)
{'cn': b'node4.example.test', 'ipHostNumber': b'192.168.1.4'}
>>> 
```